include/power_grid_model/component/fault.hpp

```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model project <dynamic.grid.calculation@alliander.com>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_COMPONENT_SHORT_CIRCUIT_HPP
+#define POWER_GRID_MODEL_COMPONENT_SHORT_CIRCUIT_HPP
+
+#include "../auxiliary/input.hpp"
+#include "../auxiliary/output.hpp"
+#include "../auxiliary/update.hpp"
+#include "../calculation_parameters.hpp"
+#include "base.hpp"
+
+namespace power_grid_model {
+
+class Fault final : public Base {
+   public:
+    using InputType = FaultInput;
+    using UpdateType = FaultUpdate;
+    template <bool sym>
+    using OutputType = FaultOutput;
+    static constexpr char const* name = "fault";
+    ComponentType math_model_type() const final {
+        return ComponentType::fault;
+    }
+
+    Fault(FaultInput const& fault_input)
+        : Base{fault_input},
+          fault_object_{fault_input.fault_object},
+          r_f_{is_nan(fault_input.r_f) ? (bool)0.0 : fault_input.r_f},
+          x_f_{is_nan(fault_input.x_f) ? (bool)0.0 : fault_input.x_f} {
+    }
+
+    FaultCalcParam calc_param(double const& u_rated, bool const& is_connected_to_source = true) const {
+        if (!energized(is_connected_to_source)) {
+            return FaultCalcParam{};
+        }
+        // param object
+        FaultCalcParam param{};
+        // calculate the fault admittance in p.u.
+        double const base_y = base_power_3p / u_rated / u_rated;
+        DoubleComplex y_f = 1.0 / (r_f_ + 1.0i * x_f_) / base_y;
```

Comment:
Can you do a research and/or experiment to be sure that if you do `1.0 / z`  where `z` is a zero complex number, the result is indeed `inf` for both real and imaginary part?

Also check if a `inf / base_y` also results in `inf`.

Reply:
I have Googled a bit. The behaviour is not actually consistant. Sometimes you get `NaN`, sometimes you get `inf`. Maybe it is good to define `z_f` in the math input? And we can check if `z_f == 0.0` in the core to decide our operation.

Reply:
implemented it in this function instead